source/02_algo/imitation_learning_en.rst

```diff
@@ -0,0 +1,81 @@
+Adversarial Structured IL. The following is a brief analysis of each direction:
+
+
+
+Research Direction
+--------
+
+BC
+~~~~~~~~
+
+BC was first proposed in [1], which proposes a supervised learning method, which directly establishes the state-action mapping relationship by fitting expert data.
+
+The biggest advantage of BC is that it is very efficient and the algorithm is simple, but once the agent encounters a state it has never seen before, it may make the wrong behavior - this problem is called "state drift". In order to solve this problem, the DAgger [2] method adopts a method of dynamically updating the data set, and continuously adds new expert data to the data set according to the real state encountered by the training policy. In the follow-up research, IBC [3] adopted the method of implicit behavior cloning, the key of which is to train a neural network to accept observations and actions, and output a number, which is low for expert actions high for non-expert actions, thus turning behavioral cloning into an energy-based modeling problem.
+
+The current research hotspots of BC algorithms mainly focus on two aspects: meta-learning and behavior cloning using VR devices.
+
+
+IRL
+~~~~~~~~
+
+The main goal of IRL is to solve the problem of finding enough high-quality data during data collection. Specifically, IRL first learns a reward function from expert data, and then uses this reward function for subsequent RL training. With such an approach, IRL can theoretically outperform expert data.
+
+From the specific work above, Ziebart et al. [4] first proposed the maximum entropy IRL, which utilizes the maximum entropy distribution to obtain good prospects and effective optimization. Later in 2016, Finn et al. [5] proposed a model-based approach to IRL called Guided Cost Learning (guided cost learning), this method uses a neural network to represent the cost to improve the expressive ability. Subsequently, Hester et al. proposed DQfD [6], which requires only a small amount of expert data, and significantly speeds up training through the pre-training startup process and subsequent learning process. Later methods such as T-REX [7] proposed a structure based on ranking expert data, which indirectly learned the reward function by comparing which expert data performed better.
+
+
+Adversarial Structured IL
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The main goal of the Adversarial Structured IL approach is to address the efficiency of IRL. It can be seen from the algorithm of IRL that even if it learns a very good reward function, since the final strategy still needs to perform reinforcement learning steps, if the strategy can be learned directly from expert data, the efficiency can be greatly improved. Based on this idea GAIL
+[8] combines generative network GAN and maximum entropy IRL, which can be continuously and efficiently trained without the need to manually label expert data continuously.
```

Comment:
The main goal of the Adversarial Structured IL approach is to improve the efficiency of IRL. Even if the IRL algorithm learns a very good reward function, in order to get the final near-optimal policy, it still needs to perform a reinforcement learning step. If the policy can be learned directly from the expert data, the efficiency can be greatly improved. Based on this idea, GAIL [8] combines generative network (GAN) and maximum entropy IRL to learn approximate optimal policies without human annotated expert data.